### ISSUE TITLE
add support for RFC 6186 for server autoconfiguration

### ISSUE THREAD

**Issue by monperrus** [deltachat/deltachat-android#1367]
Many users do not know the IMAP/SMTP servers of their email provider.

[RFC 6186, Use of SRV Records for Locating Email Submission/Access Services](https://tools.ietf.org/html/rfc6186) solves this problem, by using DNS SRV records to provide this information.

I don't see RFC 6186 in [the list of supported standards by deltachat](https://github.com/deltachat/deltachat-android/blob/master/standards.md)

What about adding support for RFC 6186 for server autoconfiguration?

**Comment by EchedeyLR** [comment]
Was not Autoconfigure already for that?
Well, Idk about this, just asking.

**Comment by monperrus** [comment]
Thuderbird's style autoconfigure (https://developer.mozilla.org/en-US/docs/Mozilla/Thunderbird/Autoconfiguration) addresses the same problem, but this is not a standard.

**Comment by monperrus** [comment]
For the record, support of RFC 6186 has been added in K9: https://github.com/k9mail/k-9/pull/4719

**Comment by EchedeyLR** [comment]
Thank you for the info. Let's wait for devs to see

**Comment by r10s** [comment]
> Thuderbird's style autoconfigure [...] addresses the same problem, but this is not a standard.

well although not being an RFC, it is widely used, isn't it? :)

however, autoconfigure is not good at custom-domains, is that better handled by RFC 6186 ? so if you have a custom-email-address at, say gmail.com or outlook? and is it actually used by at least providers offering custom-domains?

i'd say that would be a good reason to push that forward :)

**Comment by EchedeyLR** [comment]
> > Thuderbird's style autoconfigure [...] addresses the same problem, but this is not a standard.
> 
> well although not being an RFC, it is widely used, isn't it? :)
> 
> however, autoconfigure is not good at custom-domains, is that better handled by RFC 6186 ? so if you have a custom-email-address at, say gmail.com or outlook? and is it actually used by at least providers offering custom-domains?
> 
> i'd say that would be a good reason to push that forward :)

Maybe sections 3.4 and 4 from the RFC specification could help to understand?

**Comment by monperrus** [comment]
yes, some email providers for custom domains support RFC 6186, such as Axigen, see https://www.axigen.com/documentation/dns-based-service-discovery-p3277117

**Comment by angelo-fuchs** [comment]
Should this issue be moved to core? Then we would have that for all platforms at once.

**Comment by EchedeyLR** [comment]
> Should this issue be moved to core? Then we would have that for all platforms at once.

I agree with it but must be someone with permissions to do it.
